Tasks/SonarQubePostTest/SonarQubeMetrics.ps1

```diff
@@ -0,0 +1,168 @@
+#region Public 
+
+#
+# Gets the chached quality gate status or queries the server to get the result of the quality gate if it is not cached
+# Remark: this only works for SQ version 5.3+
+#
+function GetOrFetchQualityGateStatus
+{
+    $qualityGateStatus = GetTaskContextVariable "MSBuild.SonarQube.QualityGateStatus"
+    
+    if ([String]::IsNullOrEmpty($qualityGateStatus))
+    {
+        $qualityGateStatus = FetchQualityGateStatus  
+        SetTaskContextVariable "MSBuild.SonarQube.QualityGateStatus" $qualityGateStatus              
+    }
+    
+    Assert (![String]::IsNullOrEmpty($qualityGateStatus)) "Could not fetch the quality gate status"
+    
+    return $qualityGateStatus
+}
+
+#endregion
+
+#region Private
+
+function FetchQualityGateStatus
+{
+    $analysisId = GetTaskContextVariable "MSBuild.SonarQube.AnalysisId"       
```

Comment:
I am also caching and exposing the analysis ID in a build variable. Users may now use the analysis ID to fetch other information not just the quality gate